Indicators/MassIndex.cs

```diff
@@ -63,28 +85,17 @@ public override void Reset()
         /// </returns>
         protected override decimal ComputeNextValue(TradeBar input)
         {
-            _ema.Update(new IndicatorDataPoint
+            _ema1.Update(input.Time, input.High - input.Low);
+            if (_ema2.IsReady)
             {
-                Time = input.Time,
-                Value = input.High - input.Low
-            });
-
-            _ema2.Update(_ema.Current);
-
-            _sum.Update(new IndicatorDataPoint
-            {
-                Time = input.Time,
-                Value = _ema.Current / _ema2.Current
-            });
+                _sum.Update(input.Time, _ema1.Current / _ema2.Current);
```

Comment:
This introduced a lag of `period` - is it covered by a test? Something to support case for introducing lag?

Reply:
The unit test was hiding this because when the value in the target column is empty, the test continues:
[TestHelper.cs#L110](https://github.com/QuantConnect/Lean/blob/master/Tests/Indicators/TestHelper.cs#L110)
[TestHelper.cs#L208](https://github.com/QuantConnect/Lean/blob/master/Tests/Indicators/TestHelper.cs#L208)
[TestHelper.cs#L261](https://github.com/QuantConnect/Lean/blob/master/Tests/Indicators/TestHelper.cs#L261)